### PR TITLE
Master

### PR DESCRIPTION
**Pull request by kadimgh**
Implemented support of armcc5 compiler warnings and correspondent tests

### CODE REVIEW

**Comment by uhafner** [comment]
Woludn't it make more sense to have just a second parser that uses the same name in the user interface? So a user does not need to worry aboout the version. 

I.e., you use the same static variables for the name, link and trend graph in your new parser class. Then you can override the isLineIsInteresting method e.g. to detect the hash # symbol in your case.

File: src/main/java/hudson/plugins/warnings/parser/Armcc5CompilerParser.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+public class Armcc5CompilerParser extends RegexpLineParser {
+    private static final long serialVersionUID = -2677728927938443701L;
+
+    private static final String ARMCC5_WARNING_PATTERN = "^(.+)\\((\\d+)\\): (warning|error):  #(.+): (.+)$";
+    
+    /**
+     * Creates a new instance of {@link Armcc5CompilerParser}.
+     */
+    public Armcc5CompilerParser() {
+        super(Messages._Warnings_Armcc5_ParserName(),
```
Comment (uhafner) [inline src/main/java/hudson/plugins/warnings/parser/Armcc5CompilerParser.java]: Messages._Warnings_Armcc_ParserName and so on...

File: src/main/java/hudson/plugins/warnings/parser/Armcc5CompilerParser.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                Messages._Warnings_Armcc5_LinkName(),
+                Messages._Warnings_Armcc5_TrendName(),
+            ARMCC5_WARNING_PATTERN);
+    }
+
+    @Override
```
Comment (uhafner) [inline src/main/java/hudson/plugins/warnings/parser/Armcc5CompilerParser.java]: This method should not be overrided for new parsers.

File: src/main/resources/hudson/plugins/warnings/parser/Messages.properties [hunk 3]
```diff
@@ -56,6 +56,10 @@ Warnings.Armcc.ParserName=Armcc Compiler
 Warnings.Armcc.LinkName=Armcc Compiler Warnings
 Warnings.Armcc.TrendName=Armcc Compiler Warnings Trend
 
+Warnings.Armcc5.ParserName=Armcc5 Compiler
```
Comment (uhafner) [inline src/main/resources/hudson/plugins/warnings/parser/Messages.properties]: Not required.

**Comment by kadimgh** [comment]
> Woludn't it make more sense to have just a second parser that uses the same name in the user interface? So a user does not need to worry aboout the version.

I made it similar to gcc3/gcc4. They use different parser for different versions. I don't feel that hiding parsers under single name is the right way to handle different compiler versions.

> override the isLineIsInteresting

Done

> This method should not be overrided for new parsers.

Done

**Comment by uhafner** [comment]
The reason that there are two different IDs for gcc is that they will partly pick the same warnings (which results in some duplications). In your case the correct parser will be selected by the isLineIsInteresting method. No duplication will occur and the user has a simpler drop down menu. Most of the parsers are modeled in this way (Java parser, etc.). (Otherwise we would have a much longer drop down menu).

But I will leave it up to you. Just one note: when making the decision rather consider the mental model of the user and not the architectural implementation.

**Comment by kadimgh** [comment]
@uhafner not sure why Jenkins check failed and how do I restart it

**Comment by uhafner** [comment]
Works locally. Thanks for adding support for the new parser!

**Pull request merged**